platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/binding/post/PostResponseCreator.java

```diff
@@ -79,7 +80,10 @@ public Response getSamlpResponse(
     submitFormUpdated = submitFormUpdated.replace("{{" + Idp.SAML_TYPE + "}}", "SAMLResponse");
     submitFormUpdated =
         submitFormUpdated.replace("{{" + Idp.SAML_RESPONSE + "}}", encodedSamlResponse);
-    submitFormUpdated = submitFormUpdated.replace("{{" + Idp.RELAY_STATE + "}}", relayState);
+    submitFormUpdated =
+        StringUtils.isNotBlank(relayState)
+            ? submitFormUpdated.replace("{{" + Idp.RELAY_STATE + "}}", relayState)
+            : submitFormUpdated.replace("{{" + Idp.RELAY_STATE + "}}", "");
```

Comment:
✏️ This is nitpicky, but I think we prefer to use `String.format` over String concatentation (`String + String`) for minor performance reasons.

Reply:
Actually I think the reverse may be true. https://stackoverflow.com/questions/513600/should-i-use-javas-string-format-if-performance-is-important

Reply:
@brendan-hofmann you're right, I know that we tend to prefer `String.format` over concatenation but I was wrong as to why. From discussing with others, apparently it's because it's cleaner and shouldn't matter performance-wise in most cases. With 2 or 3 concatenations, it's probably fine. @blen-desta it's up to you what you want to do 👍